packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1557,6 +1557,69 @@ export function hasInstanceAffectedParent(
 // either cached the font or preloaded it earlier.
 const SUSPENSEY_FONT_TIMEOUT = 500;
 
+function customizeViewTransitionError(error: Object): mixed {
+  if (typeof error === 'object' && error !== null) {
+    switch (error.name) {
+      case 'TimeoutError': {
+        // We assume that the only reason a Timeout can happen is because the Navigation
+        // promise. We expect any other work to either be fast or have a timeout (fonts).
+        if (__DEV__) {
+          // eslint-disable-next-line react-internal/prod-error-codes
+          return new Error(
+            'A ViewTransition timed out because a Navigation stalled. ' +
+              'This can happen if a Navigation is blocked on React itself. ' +
+              "Such as if it's resolved inside useEffect. " +
+              'This can be solved by moving the resolution to useLayoutEffect.',
```

Comment:
I'm not following this one. Any chance we can manually trigger this scenario in the fixture?

Reply:
If this switches to useEffect instead of useLayoutEffect it should trigger it.

https://github.com/facebook/react/blob/main/fixtures/view-transition/src/components/App.js#L84-L86

Since the navigation should be blocked by the commit and the navigation blocks the commit from preceding the timing of unblocking the navigation needs to be in useLayoutEffect.